Modules/Scripted/ShapeAnalysisToolBox/DataImporter.py

```diff
@@ -98,41 +137,146 @@ def findWidget(self, widget, objectName):
+      if self.labelRangeInCohort != (-1, -1) and labelRange != self.labelRangeInCohort:
+        print 'ERROR: Number of labels do not match in the cohort'
+        return
+
+      self.labelRangeInCohort = labelRange
 
       # extract just the filename, and populate the tablewidget
       rowPosition = self.SubjectsTableWidget.rowCount
       self.SubjectsTableWidget.insertRow(rowPosition)
       self.SubjectsTableWidget.setItem(rowPosition, 0, qt.QTableWidgetItem(fileName))
       numCases = numCases + 1
-      print 'Columnt count is: ' + str(self.SubjectsTableWidget.columnCount)
-      print 'Row size is: ' + str(self.SubjectsTableWidget.rowCount)
-      # given labels, and current mode fill the structure table
+    self.SubjectsTableWidget.setCurrentCell(0, 0)
+
+    print 'Cohort label range is: ' + str(self.labelRangeInCohort)
+    # given labels, and current mode populate the structures slider
+    self.labelRangeInCohort = (int(self.labelRangeInCohort[0]), int(self.labelRangeInCohort[1]))
+    self.StructuresSliderWidget.minimum = int(self.labelRangeInCohort[0])
+    self.StructuresSliderWidget.maximum = int(self.labelRangeInCohort[1])
+    self.StructuresSliderWidget.setValue(0)
+
+    # Populate the topology table
+    self.populateTopologyDictionary()
+
+    self.onSubjectTableWidgetClicked(0,0)
+
+  #
+  # Function to estimate topology of segmentations, and check for consistencies.
+  # TODO: Should move to logic
+  #
+  def populateTopologyDictionary(self):
 
+    # Create vtk objects that will be used to clean the geometries
+    polydataCleaner = vtk.vtkCleanPolyData()
+    connectivityFilter = vtk.vtkPolyDataConnectivityFilter()
+    extractEdgeFilter = vtk.vtkExtractEdges()
+
+    firstValue = {} # This will be used for checking consistent, inconsistent topologies
+
+    for nodeName in self.testCaseDict.keys():
+      # Topology table is a dictionary of dictionaries.
+      self.topologyDict[nodeName] = {}
+      for segmentNum in range(self.labelRangeInCohort[0], self.labelRangeInCohort[1]+1):
+        # 0 label is assumed to be the background.
+        if segmentNum == 0:
+          continue
+        segmentId = str(segmentNum)
+        polydata = self.segmentationDict[nodeName].GetClosedSurfaceRepresentation(segmentId)
+        if polydata == None:
+          print 'Ignoring segment id ' + segmentId + ' for case: ' + nodeName
+          continue
+        # clean up polydata
+        polydataCleaner.SetInputData(polydata)
+        polydataCleaner.Update()
+        cleanData = polydataCleaner.GetOutput()
+
+        # Get the largest connected component
+        connectivityFilter.SetInputData(cleanData)
+        connectivityFilter.SetExtractionModeToLargestRegion()
+        connectivityFilter.SetScalarConnectivity(0)
+        connectivityFilter.Update()
+        largestComponent = connectivityFilter.GetOutput()
+
+        # Clean the largest component to get rid of spurious points
```

Comment:
I am thinking about this a bit more. I think we have to inform the user about the fact the segmentation was changed and cleaned, thus not exactly the same than the input he used.